### PR TITLE
Prevent mobile browser keyboard popping up when tapping on the date range input

### PR DESCRIPTION
**Pull request by gitjul**
The solution is to have two alternative, identical elements:
- a button displayed for "mobile" viewports
- the regular input displayed for wider viewports

### CODE REVIEW

File: app/assets/stylesheets/global/forms.css.scss [hunk 1]
```diff
@@ -14,11 +14,9 @@ input {
 
   &[disabled] {
     opacity: 0.4;
-    transition: opacity 0.5s;
```
Comment (gitjul) [inline app/assets/stylesheets/global/forms.css.scss]: simply not needed

File: app/assets/stylesheets/modules/buttons.css.scss [hunk 2]
```diff
@@ -15,6 +15,24 @@
   color: white;
 }
 
+.button--input {
```
Comment (gitjul) [inline app/assets/stylesheets/modules/buttons.css.scss]: styles that make the button resemble the input exactly

File: app/assets/stylesheets/modules/filters.css.scss [hunk 3]
```diff
@@ -78,3 +78,15 @@
   text-decoration: underline !important;
   margin: 0 !important;
 }
+
```
Comment (gitjul) [inline app/assets/stylesheets/modules/filters.css.scss]: show or hide button or input depending on viewport

File: app/javascript/elm/src/TimeRange.elm [hunk 4]
```diff
@@ -59,15 +60,19 @@ view refreshTimeRange status tooltipIcon resetIcon =
         [ input
             [ id "time-range"
             , attribute "autocomplete" "off"
-            , class "input-dark"
-            , class "input-filters"
-            , class "input-time"
-            , placeholder "time frame"
-            , type_ "text"
+            , class "input-dark input-filters input-time"
```
Comment (gitjul) [inline app/javascript/elm/src/TimeRange.elm]: this works exactly the same

File: app/javascript/elm/tests/TimeRangeTests.elm [hunk 5]
```diff
@@ -50,11 +51,11 @@ all =
                 in
                 TimeRange.update TimeRange.defaultTimeRange value
                     |> Expect.equal expected
-        , test "viewTimeFilter has a button" <|
+        , test "viewTimeFilter has a reset time frame button" <|
             \_ ->
                 TimeRange.view Msg Active defaultIcon defaultIcon
                     |> Query.fromHtml
-                    |> Query.find [ tag "button" ]
```
Comment (gitjul) [inline app/javascript/elm/tests/TimeRangeTests.elm]: there are two button elements there now

File: app/javascript/javascript/filtersUtils.js [hunk 6]
```diff
@@ -21,35 +21,42 @@ export const presentMoment = () =>
     .format("X");
 
 export const oneHourAgo = () =>
-  moment()
+  moment
     .utc()
     .subtract(1, "hour")
     .format("X");
 
+const humanTime = time =>
```
Comment (gitjul) [inline app/javascript/javascript/filtersUtils.js]: 😏

File: app/javascript/elm/src/TimeRange.elm [hunk 7]
```diff
@@ -1,5 +1,6 @@
 module TimeRange exposing (TimeRange(..), defaultTimeRange, update, view)
 
+import Browser.Dom as Dom
```
Comment (AkkarinForest) [inline app/javascript/elm/src/TimeRange.elm]: not needed

**Pull request merged**